### ISSUE TITLE
NotImplementedError: UserDefinedObjectVariable(ABCMeta) is not a constant

### ISSUE THREAD

**Issue by wconstab** [pytorch/torchdynamo#221]
## Summary
With abstract module classes, _setattr_ doesn't work under dynamo.
* seems specific to ABC classes, error disappears for plain classes
* also seems specific to _setattr_ method- a simple 'foo' method works fine

## Repro

```
from typing import Any, List
import torch
import torchdynamo
from abc import ABC, abstractmethod

class Metric(torch.nn.Module, ABC):
    def __setattr__(self, name: str, value: Any) -> None:
        super().__setattr__(name, value)
    
    def forward(self, *args: Any, **kwargs: Any) -> Any:
        pass

def toy_example(a, b):
    met = Metric()
    met.foo = 0

def noop_compiler(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]):
    return gm.forward

with torchdynamo.optimize(noop_compiler):
    toy_example(torch.randn(10), torch.randn(10))
```

## Logs
```
python example.py 
ERROR FROM offset=10 filename example.py 9 NotImplementedError
========== TorchDynamo Stack Trace ==========
Traceback (most recent call last):
  File "/home/whc/pytorch/torchdynamo/torchdynamo/convert_frame.py", line 174, in _convert_frame_assert
    code = transform_code_object(frame.f_code, transform)
  File "/home/whc/pytorch/torchdynamo/torchdynamo/bytecode_transformation.py", line 338, in transform_code_object
    transformations(instructions, code_options)
  File "/home/whc/pytorch/torchdynamo/torchdynamo/convert_frame.py", line 150, in transform
    tracer.run()
  File "/home/whc/pytorch/torchdynamo/torchdynamo/symbolic_convert.py", line 300, in run
    and self.step()
  File "/home/whc/pytorch/torchdynamo/torchdynamo/symbolic_convert.py", line 278, in step
    getattr(self, inst.opname)(inst)
  File "/home/whc/pytorch/torchdynamo/torchdynamo/symbolic_convert.py", line 142, in wrapper
    return inner_fn(self, inst)
  File "/home/whc/pytorch/torchdynamo/torchdynamo/symbolic_convert.py", line 583, in CALL_FUNCTION
    self.call_function(fn, args, {})
  File "/home/whc/pytorch/torchdynamo/torchdynamo/symbolic_convert.py", line 217, in call_function
    self.push(fn.call_function(self, args, kwargs))
  File "/home/whc/pytorch/torchdynamo/torchdynamo/variables/misc.py", line 439, in call_function
    return self.obj.call_method(tx, self.name, args, kwargs).add_options(self)
  File "/home/whc/pytorch/torchdynamo/torchdynamo/variables/misc.py", line 63, in call_method
    inner_fn = self.const_getattr(self, name)
  File "/home/whc/pytorch/torchdynamo/torchdynamo/variables/misc.py", line 38, in const_getattr
    search_type = self.typevar.as_python_constant()
  File "/home/whc/pytorch/torchdynamo/torchdynamo/variables/base.py", line 137, in as_python_constant
    raise NotImplementedError(f"{self} is not a constant")
NotImplementedError: UserDefinedObjectVariable(ABCMeta) is not a constant
========== Exception (above) while processing ==========
  File "example.py", line 24, in <module>
    toy_example(torch.randn(10), torch.randn(10))
  File "example.py", line 16, in toy_example
    def toy_example(a, b):
  File "/home/whc/pytorch/torch/nn/modules/module.py", line 260, in __init__
    self.training = True
  File "example.py", line 8, in __setattr__
    def __setattr__(self, name: str, value: Any) -> None:
========== End debug info ==========
```

**Comment by jansel** [comment]
Thanks for the repro!  I think ABC might be confusing the inlining logic.

**Comment by jansel** [comment]
@wconstab is this fixed by #227?  Or still an issue?
